sdk/python/feast/data_source.py

```diff
@@ -670,6 +671,7 @@ def __init__(
         query: Optional[str] = None,
     ):
         self._bigquery_options = BigQueryOptions(table_ref=table_ref, query=query)
+        self._assert_table_exists()
```

Comment:
External validations like these should not happen in the constructor. Can you move it to happen when the user runs `feast apply`?